accounts-db/src/accounts_db.rs

```diff
@@ -5752,7 +5752,7 @@ impl AccountsDb {
     fn has_space_available(&self, slot: Slot, size: u64) -> bool {
         let store = self.storage.get_slot_storage_entry(slot).unwrap();
         if store.status() == AccountStorageStatus::Available
-            && (store.accounts.capacity() - store.accounts.len() as u64) > size
+            && store.accounts.remaining_bytes() > size
```

Comment:
Since we are fixing `remaining bytes`, `has_space_available` should be fixed too.